course/templates/course/flow-page.html

```diff
@@ -284,6 +284,55 @@
 
   {# }}} #}
 
+
+  {# {{{ flow page interaction email #}
+
+  {% if may_send_email_about_flow_page %}
+      <div class="row" align="right">
+          <a href="{% url 'relate-flow_page_interaction_email' course.identifier flow_session.id ordinal %}"
+             class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+             title="{% trans 'Send an interaction email to course staff about this page' %}"
+             target="_blank"
+          >
+          {% trans "Send email" %}
+      </a>
+      </div>
+  {% endif %}
+
+  {# }}} #}
+
+
+  {# {{{ show single grade link #}
+
+  {% if show_privileged_shortcut %}
+    <div class="row" align="right">
+       {% if not flow_session.in_progress %}
+            <a href="{% url 'relate-view_reopen_session' course.identifier flow_session.id opportunity.id %}"
+               class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+               target="_blank"
+            >
+              {% trans "Reopen" %}
+            </a>
+        {% endif %}
+          <a href="{% url 'relate-grant_exception_stage_3' course.identifier flow_session.participation.id flow_session.flow_id flow_session.id %}"
+             class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+             target="_blank"
+          >
+              {% trans "Grant exception" %}
+          </a>
+
+      <a class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+         href="{% url 'relate-grade_flow_page' course.identifier flow_session.id ordinal %}"
+         role="button"
+         target="_blank"
+      >
+        {% trans "View in grading page" %}
```

Comment:
Sorry that these comments come out so piece-by-piece, but I only just discovered (ahem) this bit of the PR. :)

The permissions for this need to be a bit finer-grained.

* Reopen is `pperm.reopen_flow_session`.
* Grant exception is `pperm.grant_exception`.
* Seeing the grading page is `pperm.view_gradebook`.

You actually don't need to pass a flag from Python to the template. Just

```
{% if pperm.XYZ %}
{% endif %}
```
should do the trick.